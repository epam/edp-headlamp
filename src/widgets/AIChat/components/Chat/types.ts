import { CodemieSecretData, ConversationItem } from '../../types';

export interface ChatProps {
  codemieSecretData: CodemieSecretData;
  conversation: ConversationItem;
  updateConversation: (conversation: ConversationItem) => void;
  requestError: Error;
}
