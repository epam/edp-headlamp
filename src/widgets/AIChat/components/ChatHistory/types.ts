import { ConversationItem, ConversationState } from '../../types';

export interface ChatHistoryProps {
  conversations: ConversationItem[];
  activeConversation: ConversationItem | null;
  setActiveConversation: React.Dispatch<React.SetStateAction<ConversationItem | null>>;
  setConversations: React.Dispatch<React.SetStateAction<ConversationState>>;
  createNewConversation: () => void;
}
