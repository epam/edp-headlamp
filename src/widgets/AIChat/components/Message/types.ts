import React from 'react';
import { ChatEntity } from '../../types';

export interface MessageProps {
  entityRole: ChatEntity;
  createdAt: string;
  content: React.ReactNode | string;
}
