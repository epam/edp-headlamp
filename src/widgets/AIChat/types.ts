import { ValueOf } from '../../types/global';
import { CHAT_ENTITY } from './constants';

export type ChatEntity = ValueOf<typeof CHAT_ENTITY>;

export interface ChatItemRequest {
  message: string;
}

export interface ProcessedChunk {
  id: string;
  text: string;
  isCode: boolean;
}

export interface ChatItemResponse {
  message: string | undefined; // is available on streaming finish
  processedChunks: ProcessedChunk[];
}

export interface HistoryStateItem {
  id: string;
  createdAt: string;
  request: ChatItemRequest;
  response?: ChatItemResponse;
}

export interface HistoryPayloadItem {
  createdAt: string;
  message: string;
  role: ChatEntity;
}

export interface ConversationItem {
  conversationId: string;
  conversationHistory: HistoryStateItem[];
  assistantID: string;
  assistantName: string;
  assistantContext: string;
  prompt: string;
  llmModel: string;
  assistantMode: boolean;
  file: File | null;
  stream: boolean;
  _debug: boolean;
  topK: number;
  codeMode?: boolean;
  kbMode?: boolean;
  kbType?: string;
  kbName?: string;
  appName?: string;
  repoName?: string;
  indexType?: string;
}

export interface ConversationPayload {
  conversationId: string;
  text: string;
  file: File | null;
  prompt: string;
  history: HistoryPayloadItem[];
  llmModel: string;
  stream: boolean;
  _debug: boolean;
  topK: number;
  code_fields?: {
    appName: string;
    repoName: string;
    indexType: string;
  };
}

export interface ConversationResponseChunk {
  time_elapsed: number;
  input_tokens_used: number;
  tokens_used: number;
  generated_chunk: string;
  generated: string;
  thought: string;
  last: boolean;
  debug: null;
}

export interface ConversationState {
  [key: string]: ConversationItem;
}

export interface CodemieSecretData {
  apiUrl: string;
  assistantId: string;
  token: string;
}
