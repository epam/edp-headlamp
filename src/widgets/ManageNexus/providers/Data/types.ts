import { ManageNexusCIProps } from '../../types';

export interface DataContextProviderValue extends ManageNexusCIProps {}

export interface DataContextProviderProps extends ManageNexusCIProps {}
