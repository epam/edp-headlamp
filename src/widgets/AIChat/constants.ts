export const CHAT_ENTITY = {
  USER: 'User',
  ASSISTANT: 'Assistant',
} as const;

export const GPT_4 = 'GPT_4';

export const SUPPORTED_LANGUAGES = [
  'abap',
  'abnf',
  'actionscript',
  'ada',
  'agda',
  'al',
  'antlr4',
  'apacheconf',
  'apex',
  'apl',
  'applescript',
  'aql',
  'arduino',
  'arff',
  'asciidoc',
  'asm6502',
  'aspnet',
  'autohotkey',
  'autoit',
  'bash',
  'basic',
  'batch',
  'bbcode',
  'birb',
  'bison',
  'bnf',
  'brainfuck',
  'brightscript',
  'bro',
  'bsl',
  'c',
  'cil',
  'clike',
  'clojure',
  'cmake',
  'coffeescript',
  'concurnas',
  'cpp',
  'csharp',
  'csp',
  'css',
  'css-extras',
  'cypher',
  'd',
  'dart',
  'dax',
  'dhall',
  'diff',
  'django',
  'dns-zone-file',
  'docker',
  'ebnf',
  'editorconfig',
  'eiffel',
  'ejs',
  'elixir',
  'elm',
  'erb',
  'erlang',
  'etalon',
  'euphoria',
  'exascal',
  'factor',
  'firebird',
  'flow',
  'fortran',
  'ftl',
  'futhark',
  'gcode',
  'gdscript',
  'gedcom',
  'gherkin',
  'git',
  'glsl',
  'gml',
  'go',
  'graphql',
  'groovy',
  'haml',
  'handlebars',
  'haskell',
  'haxe',
  'hcl',
  'hlsl',
  'hsqml',
  'http',
  'ichigojam',
  'icon',
  'iecst',
  'ignore',
  'inform7',
  'ini',
  'io',
  'j',
  'java',
  'javadoc',
  'javadoclike',
  'javascript',
  'javastacktrace',
  'jolie',
  'jq',
  'jsdoc',
  'jsx',
  'js-extras',
  'json',
  'jsonp',
  'json5',
  'julia',
  'keyman',
  'kotlin',
  'latex',
  'latte',
  'less',
  'lex',
  'lilypond',
  'liquid',
  'lisp',
  'livescript',
  'llvm',
  'lolcode',
  'lua',
  'makefile',
  'markup',
  'markup-templating',
  'matlab',
  'mdx',
  'mel',
  'mizar',
  'mongodb',
  'monkey',
  'moonscript',
  'n1ql',
  'n4js',
  'nand2tetris-hdl',
  'naniscript',
  'nasm',
  'neon',
  'nesc',
  'netlinx',
  'nginx',
  'nim',
  'nix',
  'nsis',
  'objectivec',
  'ocaml',
  'opencl',
  'oz',
  'parigp',
  'parser',
  'pascal',
  'pascaligo',
  'pcaxis',
  'peoplecode',
  'perl',
  'pgsql',
  'php',
  'php-extras',
  'phpdoc',
  'plsql',
  'powerquery',
  'powershell',
  'processing',
  'prolog',
  'promql',
  'properties',
  'protobuf',
  'pug',
  'puppet',
  'pure',
  'purebasic',
  'purescript',
  'python',
  'q',
  'qml',
  'r',
  'racket',
  'rado',
  'raku',
  'reason',
  'regex',
  'renpy',
  'rest',
  'rip',
  'ris',
  'roboconf',
  'robot',
  'ruby',
  'rust',
  'sas',
  'sass',
  'scss',
  'scala',
  'scheme',
  'scss',
  'shell-session',
  'smalltalk',
  'smarty',
  'sml',
  'solidity',
  'solution-file',
  'soy',
  'sparql',
  'sqf',
  'sql',
  'stan',
  'stylus',
  'swift',
  't4-cs',
  't4-vb',
  't4-templating',
  'tap',
  'tcl',
  'testy',
  'textile',
  'thrift',
  'toml',
  'tsx',
  'tt2',
  'turtle',
  'twig',
  'typescript',
  'vala',
  'vbnet',
  'velocity',
  'verilog',
  'vhdl',
  'vim',
  'visual-basic',
  'warpscript',
  'wasm',
  'wenyan',
  'wiki',
  'xeora',
  'xml-doc',
  'xojo',
  'xquery',
  'yaml',
  'yang',
  'zig',
];
