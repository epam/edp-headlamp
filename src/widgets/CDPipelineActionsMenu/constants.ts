import { CDPipelineKubeObject } from '../../k8s/groups/EDP/CDPipeline';
import { CDPipelineKubeObjectConfig } from '../../k8s/groups/EDP/CDPipeline/config';

export const widgetPermissionsToCheck = {
  create: [],
  update: [{ instance: CDPipelineKubeObject, config: CDPipelineKubeObjectConfig }],
  delete: [{ instance: CDPipelineKubeObject, config: CDPipelineKubeObjectConfig }],
};
