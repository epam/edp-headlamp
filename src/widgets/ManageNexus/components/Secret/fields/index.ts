export * from './Password';
export * from './User';
export * from './URL';
