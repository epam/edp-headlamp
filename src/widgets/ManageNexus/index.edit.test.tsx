/**
 * @jest-environment jsdom
 */

import { render, screen } from '@testing-library/react';
import React from 'react';
import { TestWrapper } from '../../../mocks/wrappers/default';
import { NexusCISecretMock } from '../../k8s/groups/default/Secret/mocks/nexus-ci-secret.mock';
import { SecretKubeObjectInterface } from '../../k8s/groups/default/Secret/types';
import { SYSTEM_QUICK_LINKS } from '../../k8s/groups/EDP/QuickLink/constants';
import { DEFAULT_ALLOWED_REASON } from '../../providers/Permissions/constants';
import { FORM_MODES } from '../../types/forms';
import { ManageNexus } from './index';

test('renders ManageNexus Edit component', () => {
  render(
    <TestWrapper>
      <ManageNexus
        secret={NexusCISecretMock as unknown as SecretKubeObjectInterface}
        quickLink={{
          apiVersion: 'v1.edp.epam.com/v1',
          kind: 'QuickLink',
          metadata: {
            name: SYSTEM_QUICK_LINKS.NEXUS,
            namespace: 'test-namespace',
            creationTimestamp: '',
            uid: '',
          },
          spec: {
            type: SYSTEM_QUICK_LINKS.NEXUS,
            url: 'https://test-nexus.com',
            visible: true,
            icon: '',
          },
          status: '',
        }}
        permissions={{
          create: {
            Secret: {
              allowed: true,
              reason: DEFAULT_ALLOWED_REASON,
            },
          },
          update: {
            QuickLink: {
              allowed: true,
              reason: DEFAULT_ALLOWED_REASON,
            },
            Secret: {
              allowed: true,
              reason: DEFAULT_ALLOWED_REASON,
            },
          },
          delete: {
            Secret: {
              allowed: true,
              reason: DEFAULT_ALLOWED_REASON,
            },
          },
        }}
        ownerReference={null}
        mode={FORM_MODES.EDIT}
        handleClosePanel={jest.fn()}
      />
    </TestWrapper>
  );

  const dialog = screen.getByTestId('form');
  expect(dialog).toMatchSnapshot();
});
