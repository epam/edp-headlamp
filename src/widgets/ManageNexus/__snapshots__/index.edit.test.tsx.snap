// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ManageNexus Edit component 1`] = `
<div
  data-testid="form"
>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-1sazv7p-MuiStack-root"
      >
        <div
          class="MuiStack-root css-jfdv4h-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-2"
                id="mui-2-label"
              >
                Quick Link URL
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-2"
                  name="externalUrl"
                  placeholder="Enter URL"
                  type="text"
                  value="https://test-nexus.com"
                />
                <span />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-14lo706"
                  >
                    <span>
                      Quick Link URL
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-4"
                  id="mui-4-label"
                >
                  URL
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    disabled=""
                    id="mui-4"
                    name="url"
                    placeholder="Enter URL"
                    type="text"
                    value="https://test-url.com"
                  />
                  <div
                    class="MuiStack-root css-niqf4j-MuiStack-root"
                  >
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <span />
                    </div>
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        URL
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-6"
                  id="mui-6-label"
                >
                  User
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-6"
                    name="username"
                    placeholder="Enter user name"
                    type="text"
                    value="test-username"
                  />
                  <span />
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        User
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="MuiStack-root css-jfdv4h-MuiStack-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
            >
              <div
                class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-98a9yt-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="true"
                  for="mui-8"
                  id="mui-8-label"
                >
                  Password
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-1wqgp0s-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <input
                    aria-invalid="false"
                    autocomplete="off"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                    id="mui-8"
                    name="password"
                    placeholder="Enter password"
                    type="password"
                    value="test-password"
                  />
                  <div
                    class="MuiStack-root css-niqf4j-MuiStack-root"
                  >
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <span />
                    </div>
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
                    >
                      <button
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall css-1pe4mpk-MuiButtonBase-root-MuiIconButton-root"
                        tabindex="0"
                        type="button"
                      >
                        <span />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-14lo706"
                    >
                      <span>
                        Password
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        class="MuiStack-root css-1a41st4-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButton-colorPrimary css-pdf1aj-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          undo changes
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-vw8gqz-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          save
        </button>
      </div>
    </div>
  </div>
</div>
`;
